src/_blog/aut-tls.md

```diff
@@ -0,0 +1,155 @@
+
+This blog post explores the challenges of browser-node connectivity, how AutoTLS works, and how it fits into the libp2p project and the browser ecosystem.
+
+If you're interested in trying it out, we have an example with go-libp2p and js-libp2p:
+
+- [AutoTLS example for go-libp2p](https://github.com/libp2p/go-libp2p/tree/master/examples/autotls)
+- [AutoTLS example for js-libp2p](https://github.com/libp2p/js-libp2p-example-auto-tls)
+
+## Why is browser-node connectivity hard?
+
+Historically, the Web relied solely on HTTP, which is unencrypted, undermining user privacy and exposing users to MITM attacks. With the advent of TLS, HTTPS, and automated certificate authorities like Let's Encrypt, encryption is now the norm. So much so that [Chrome warns users "Not secure"](https://blog.google/products/chrome/milestone-chrome-security-marking-http-not-secure/) when they're visiting a site over an unencrypted HTTP connection, or produce "mixed-content" errors when a page tries to fetch unencrypted subresources.
+
+![Chrome warning](../assets/autotls/computer-says-no.png)
+
+In libp2p, [all connections are encrypted](https://docs.libp2p.io/concepts/secure-comm/overview/#overview) by default, using either Noise or TLS.
+
+For as long as libp2p has existed, browser-node connectivity has been a challenge. Browser does not see Noise-encrypted WebSocket  connections as part of "[Secure Context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts)", leaving us with TLS encryption as the only viable option. 
+
+Up until recently, configuring libp2p node to be reachable from the browser required additional effort, as node operators had to own and manually configure a domain name and obtain a TLS certificate signed by a certificate authority (CA).
+
+Recent investments in [WebTransport](https://connectivity.libp2p.io/#webtransport) and [WebRTC](https://connectivity.libp2p.io/#webrtc) helped circumvent this problem, by removing the need for CA-signed TLS certificate, but they have their own drawbacks outlined below.
+
+Experience has shown that WebSockets are still the most common and reliable way to establish a bi-directional streaming connection from a browser. That's not to say that WebSockets are perfect. Most notably, in libp2p, [Secure WebSockets require 6 round trips to establish a connection](https://connectivity.libp2p.io/#websocket?tab=websocket-in-libp2p). By comparison, [WebTransport requires 3 round trips](https://connectivity.libp2p.io/#webtransport?tab=webtransport-in-libp2p), which is why we believe WebTransport is the future of browser-node connectivity.
+
+## How AutoTLS works
+
+With AutoTLS, the end result is:
+
+1. Your libp2p node, identified by a PeerID, has a wild card certificate for `*.<PeerID>.libp2p.direct`.
```

Comment:
is it worth mentioning somewhere that `<PeerID>` is actually the [base36 representation of the peerid](https://github.com/ipshipyard/p2p-forge/blob/71cc95e9a425e55cfc9140a977a14336dd563f77/client/acme.go#L449)?

we could show an example address such as `/ip4/1.2.3.4/tcp/8888/tls/sni/1-2-3-4.k51qzi5uqu5dgarn9dk0o1grux8wt6ey0vk9xjc6oumuz8x425r5tndaqmdy6q.libp2p.direct/ws/p2p/12D3KooWA8YZg39SL7KpvFgRXeN3uYu1WpdTbo2CQmFe1us7oWVP`

Reply:
We have an example below. But I think it's worth pointing out that it's base36 encoded

Reply:
Added a note: https://github.com/libp2p/blog/pull/137/commits/87c8ebf72f5968c25bb7195cf4aefc9711962b2b